content/modules/redisgears/write-behind.md

```diff
@@ -0,0 +1,125 @@
+
+## Write-Behind Caching with RedisGears
+
+[RGSync](https://github.com/RedisGears/rgsync) is a RedisGears recipe that uses the write-behind pattern to synchronize Redis data structures to a backing data store (Oracle, MySQL, Cassandra, and Snowflake are currently supported).
+
+Here's how you can create a write-behind function using the `RGSync` recipe against MySQL. This function will synchronize a set of fields in a Redis hash to the columns in a MySQL row. The key name for the Redis hash must have the form `{name}:{id}`, where `id` is the primary key for the row we want to keep in sync with. So, for example, suppose our hashes look like this:
+
+```sh
+redis.cloud:6379> HSET person:5 first_name Marek last_name Michalski age 27
+(integer) 2
+```
+
+The data in this hash would correspond to a row with ID of 5. But we need to define the mapping from our hash field to our columns. In our function, we do this using a Python dictionary:
+
+```py
+personsMappings = {
+	'first_name':'first',
+	'last_name':'last',
+	'age':'age'
+}
+```
+
+This is saying that the `last_name` field in our hash should be synchronized with the 'last' field in our row. So, now we're ready to write our RedisGears function.
+
+First, load the necessary libraries and connect to MySQL:
+
+```py
+from rgsync import RGWriteBehind
+from rgsync.Connectors import MySqlConnector, MySqlConnection
+
+connection = MySqlConnection('myapplication', 'Password123!', 'localhost:3306/test')
+```
+
+Next, create a connector object, specifying the table name and the name of the column storing the primary key:
+
+```py
+personConnector = MySqlConnector(mySqlConnection, 'persons', 'id')
+```
+
+After that, define your field-column mappings:
+
+```py
+personsMappings = {
+	'first_name':'first',
+	'last_name':'last',
+	'age':'age'
+}
+```
+
+Finally, use these initialized objects to create an `RGWRiteBehind` object:
+
+```py
+RGWriteBehind(GB,
+              name='PersonsWriteBehind',
+              version='1',
+              connector=personsConnector,
+              keysPrefix='person',
+              mappings=personsMappings)
+```
+
+Notice here that you can version this function registration with the `version` argument, as newer versions with the same `name` will replace the old one.
+
+If you've placed all of this code into a file (`example.py`), then you can load it into your Redis deployment using the [`gears-cli`](https://github.com/RedisGears/gears-cli) tool:
+
+```sh
+gears-cli --host <host>
+          --port <post>
+          --password <password>
+          example.py
+          REQUIREMENTS rgsync
+```
+
+### Secret Management
+
+You may not want to store database credentials in your RedisGears functions. To avoid this, you can store these credentials in-memory as module parameters and then reference them from your functions.
```

Comment:
it's not really "in-memory". it should say "you can pass these credentials via module arguments set by the database administrator and query the credentials in the script via configet (link to it?)"